pkg/controllers/vscr/scrutinizepod_reconciler.go

```diff
@@ -147,13 +150,28 @@ func (s *ScrutinizePodReconciler) getHostList(pods []corev1.Pod) []string {
 }
 
 // buildScrutinizeCmdArgs returns the arguments of vcluster scrutinize command
-func (s *ScrutinizeCmdArgs) buildScrutinizeCmdArgs(vdb *v1.VerticaDB) []string {
+func (s *ScrutinizePodReconciler) buildScrutinizeCmdArgs(vdb *v1.VerticaDB) []string {
 	cmd := []string{
-		"--db-user", s.username,
-		"--hosts", strings.Join(s.hosts, ","),
+		"--db-user", s.ScrArgs.username,
+		"--hosts", strings.Join(s.ScrArgs.hosts, ","),
 		"--log-path", paths.ScrutinizeLogFile,
-		"--tarball-name", s.tarballName,
+		"--tarball-name", s.ScrArgs.tarballName,
+	}
+
+	// In order to facilitate diagnosing less recent problems,
+	// scrutinize should be able to collect an arbitrary time range of logs
+	if vmeta.GetScrutinizeLogAgeHours(s.Vscr.Annotations) != 0 {
+		// if s.Vscr.Annotations[vmeta.ScrutinizeLogAgeHours] != "" {
+		cmd = append(cmd, "--log-age-hours", strconv.Itoa(vmeta.GetScrutinizeLogAgeHours(s.Vscr.Annotations)))
+	} else {
+		if vmeta.GetScrutinizeLogAgeOldestTime(s.Vscr.Annotations) != "" {
+			cmd = append(cmd, "--log-age-oldest-time", s.Vscr.Annotations[vmeta.ScrutinizeLogAgeOldestTime])
```

Comment:
```suggestion
			cmd = append(cmd, "--log-age-oldest-time", vmeta.GetScrutinizeLogAgeOldestTime(s.Vscr.Annotations))
```

Reply:
We can manually rotate and check the time range of generated vertica.log files. But it's too long to wait because the minimum supported time range of these 3 annotations is 1 hour.